### DISCUSSION TITLE
OCPP

### DISCUSSION THREAD

**Discussion by profdraper** [netzero-labs/netzero#65] [General]
Great to see OCPP now added, works perfectly with my Wallbox Pulsar.

One consideration though: as it stands, the Wallbox uses its internal scheduling to override dependance on solar charging & is useful for (say) a full top-up to 100% off-peak overnight & where the day-time solar trickle just doesn't get there.

This particular scenario is useful in preparing for a long trip or simply balancing and calibrating the range for an LFP battery, needed every six weeks or so.

I assume as long as Netzero's OCPP is enabled in the charger, this will then it will not work overnight when the sun isn't shining?

Or perhaps there may well be some automation option for this, but can't quite see it yet.

Thanks. Great addition.

**Comment by zigam** [comment]
Your Wallbox schedule should still work after configuring OCPP, let me know if that's not working for you. The charger should have a green LED (not yellow), indicating it's not locked.

We'll add automation options to start/stop charging as well, based on time or electricity price (and eventually based on solar production). We'll add an instant Start/Stop as well, since that option seems to be removed from the Wallbox app when OCPP is configured.

**Reply by profdraper** [reply]
thanks for this, will try the scheduling & OCPP as you suggest.

- [ ] what your're suggesting sounds similar to Charge HQ) a subscription service, but also other features beyond that as well. Nice, thanks again.

**Reply by profdraper** [reply]
A little confused about the OCPP functionality on my Wallbox (with N1CT Energy meter):

1) If I run this without Netzero OCPP, it runs in Green mode (solar only) but also seems to use the PW3; seems no way to avoid this & automation for 'above 8kwH' seems to make no difference. However, does not pull energy from grid.

2) With OCPP configured, the automation for 'preserve powerwall charge' works, but (confusinging) the Wallbox draws substantial energy from the grid - irrespective of being set to 'green only' mode'.

Don't get it.

**Reply by zigam** [reply]
Thanks for the report, we’ll investigate. But it’s possible Green mode is not compatible with OCPP, unfortunately it’s common that features are disabled in OCPP mode. 

For the home usage automation: when you charge your EV, is that reflected in you home usage in the Tesla app?

**Comment by profdraper** [comment]
Thanks for this.Yes, charging EV usage is shown in the Tesla app. And yes, Eco/Green mode does work with other OCPP providers, eg ChargeHQ.Paul Draper23 Craigmont St., Tarragindi Q 41210400 282796On 22 Dec 2024, at 04:34, Ziga Mahkovec ***@***.***> wrote:﻿
Thanks for the report, we’ll investigate. But it’s possible Green mode is not compatible with OCPP, unfortunately it’s common that features are disabled in OCPP mode.
For the home usage automation: when you charge your EV, is that reflected in you home usage in the Tesla app?

**Reply by zigam** [reply]
1. If EV usage is shown in the Tesla app, then the "home usage above 8kW" should work. Can you elaborate on the automation "seems to make no difference"?

2. Regarding ChargeHQ, do you mean you can keep Wallbox's Eco-smart mode on while in OCPP mode and it works? Since ChargeHQ's documentation explicitly states Eco-smart is not supported: https://chargehq.net/kb/wallbox-configuration

**Comment by profdraper** [comment]
﻿﻿On 22 Dec 2024, at 17:13, Ziga Mahkovec ***@***.***> wrote:


If EV usage is shown in the Tesla app, then the "home usage above 8kW" should work. Can you elaborate on the automation "seems to make no difference"?Thanks for the followup. First, I wasn’t using the automation ‘above 8kw’,  In the case of using Netzero’s OCPP, I was using ‘When vehicle starts: set  backup reserve to the current state of charge (preserve Powerwall charge). And this works well in terms of not discharging the PW (certainly not the case in some of the scenarios noted here). What the perceived problems have been is that: the charger never really throttles to available solar and always runs at 7.5kw irrespective. If there is more than available power from the panels, fine, but the charger dips into grid whenever it likes. Perhaps you are right and I should try disabling any eco features and see how that behaves. 


Regarding ChargeHQ, do you mean you can keep Wallbox's Eco-smart mode on while in OCPP mode and it works? Since ChargeHQ's documentation explicitly states Eco-smart is not supported: https://chargehq.net/kb/wallbox-configurationI’ll have a look at that again, ChargeHQ has certainly been the best of the OCPPs & it throttles the Amperage up & down in line with the solar available. Am also trying CatchPower OCPP as well. None of them are really giving what I want though. So-called ‘100% solar charging’ seems problematic. My Wallbox in question has a N1CT powermeter & CT clamp fitted but still does not run 100% green as expected (also not throttling, drawing from the grid at times etc). I would have thought there should be no need for OCPP, but am working through that.Thanks again. Paul

**Reply by zigam** [reply]
I see. Netzero doesn't do any active charging automations yet (starting/stopping charging, or managing the charge rate). This is on the roadmap however.

For now, you can manage Powerwall configuration when Netzero is notified that charging started or stopped (based on integrations with Tesla Wall Connector/Tesla car, OCPP, or home usage thresholds). This allows you to avoid discharging the Powerwall, but won't have an impact on grid usage.

**Reply by profdraper** [reply]
SUCCESS. I finally got my Wallbox Pulsar solar charging as it should, but contrary to our earlier dicussion - solar charging and power managment setings turned ON in the Wallbox app are I can confirm are working perfectly in parallel with Netzero OCPP control.

This is my preferred use, because with Netxero OCPP I find the automations for 'EV charging started or stopped' so much better; the other opion with power load as the trigger is not as useful becuase it needs to get to a high threshold before it changes the battery backup reserve to stops it draining into the EV.

Problem with that scenario is that the EV charger is often dynmamically trickling from solar at much lower levels than that threshhole. With OCPP, the battery reserve is set at any power level & simply uses 'start charging' as a trigger to set the battery.

Brilliant. Thanks so much.

**Reply by zigam** [reply]
Great, glad it's working for you!

**Reply by profdraper** [reply]
a suggestion?
an automation or switch to pause battery charging?

presently the ev charger needs to wait all morning because the PW is busy charging. would be useful to disable this, charge the car first, then go about our day & then the PW can back to charging.

**Reply by zigam** [reply]
Tesla doesn't offer a way to pause Powerwall charging. However, if you system is in Self-Powered mode, house loads (which includes EV) will take precedence over Powerwall charging. So you can switch to Self-Powered mode, or use Netzero automations to switch to Self-Powered mode on a schedule.

**Comment by profdraper** [comment]
Thanks for this, will try & add to my EV automation.
